packages/showcase/src/client/simulation.tsx

```diff
@@ -1,22 +1,67 @@
-import Registry, {getCompName} from '@ui-autotools/registry';
-// TODO: make sure we get the project's React here.
+import Registry, {getCompName, IComponentMetadata} from '@ui-autotools/registry';
+import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 
-const url = new URL(document.location.href);
-const componentName = url.searchParams.get('component');
-const simulationTitle = url.searchParams.get('simulation');
+function findComponent(compName: string) {
+  const comps = Array.from(Registry.metadata.components.keys());
+  return comps.find((c) => getCompName(c) === compName);
+}
 
-const Comp = Array.from(Registry.metadata.components.keys()).find((c) =>
-  getCompName(c) === componentName
-);
+function findSimulation<T>(compMeta: IComponentMetadata<T>, simName: string) {
```

Comment:
Does findSimulation need to be generic? You are not calling it with an argument when you use it. 
You can use IComponentMetadata<any> instead